tests/unit/test_state.py

```diff
@@ -12,6 +12,20 @@
 from state import State
 
 
+def test_proxyconfig_invalid(monkeypatch: pytest.MonkeyPatch):
+    """
+    arrange: given a monkeypatched os.environ mapping that contains invalid proxy values.
+    act: when charm state is initialized.
+    assert: CharmConfigInvalidError is raised.
+    """
+    monkeypatch.setattr(state.os, "environ", {"JUJU_CHARM_HTTP_PROXY": "INVALID_URL"})
+    mock_charm = MagicMock(spec=ops.CharmBase)
+    mock_charm.config = {}
+
+    with pytest.raises(state.CharmConfigInvalidError):
+        state.State.from_charm(mock_charm)
```

Comment:
What about https://github.com/canonical/is-charms-contributing-guide/pull/74 (asserting on the error property)? (applies to other changes in this pr too)

Reply:
In this case, many of the error tests are raised through mock's side_effects, this would mean that i would essentially be checking for the test's own mocked exceptions. Would it be better if we separate this out in another scope?

Reply:
Fine, but maybe this is something to contribute to the PR that asserting on the error property does not make sense in all cases.

Reply:
Yup! Will comment :D Thank you!